### PR TITLE
Coverity fixes & removal of namespace usings in headers

### PR DESCRIPTION
**Pull request by hendrik-skubch**
Various coverity fixes and removal of using namespace std from headers

Companion PR is here:
https://github.com/rapyuta-robotics/alica/pull/25

### CODE REVIEW

**Review by gauthamm — APPROVED** [review]

File: process_manager/src/ManagedExecutable.cpp [hunk 1]
```diff
@@ -2,79 +2,92 @@
...
     this->robotEnvVariable = robotName;
 }
 
-ManagedExecutable::~ManagedExecutable() {
+ManagedExecutable::~ManagedExecutable()
+{
     // TODO: check whether the cleanup is right -> valgrind
     for (auto param : this->runningParams) {
-        free((char*) param);
+        free((char*)param);
```
Comment (abolinsky) [inline process_manager/src/ManagedExecutable.cpp:66]: c++ style cast

File: fsystem/src/FileSystem.cpp [hunk 2]
```diff
@@ -21,13 +29,14 @@ FileSystem::~FileSystem() {}
...
+std::string FileSystem::getSelfPath()
+{
     int size = 100;
     char* buff = NULL;
-    buff = (char*) malloc(size);
+    buff = (char*)malloc(size);
 
     while (1) {
-        buff = (char*) realloc(buff, size);
+        buff = (char*)realloc(buff, size);
```
Comment (abolinsky) [inline fsystem/src/FileSystem.cpp:39]: c++ style cast

File: fsystem/src/FileSystem.cpp [hunk 3]
```diff
@@ -21,13 +29,14 @@ FileSystem::~FileSystem() {}
  * Helpfull method to get the location of the currently executed executable.
  * @return The path to the running executable.
  */
-std::string FileSystem::getSelfPath() {
+std::string FileSystem::getSelfPath()
+{
     int size = 100;
     char* buff = NULL;
-    buff = (char*) malloc(size);
+    buff = (char*)malloc(size);
```
Comment (abolinsky) [inline fsystem/src/FileSystem.cpp:36]: c++ style cast

**Review by abolinsky — APPROVED** [review]

File: process_manager/src/ManagedExecutable.cpp (hunk 1 above)
Comment (hendrik-skubch) [reply process_manager/src/ManagedExecutable.cpp:66]: won't fix

File: fsystem/src/FileSystem.cpp (hunk 2 above)
Comment (hendrik-skubch) [reply fsystem/src/FileSystem.cpp:39]: won't fix

File: fsystem/src/FileSystem.cpp (hunk 3 above)
Comment (hendrik-skubch) [reply fsystem/src/FileSystem.cpp:36]: and won't fix....at least not now ;)

**Pull request merged**